### PR TITLE
Add test for attachments on Whitehall

### PR DESCRIPTION
**Pull request by adrianclay**
This gives confidence that Whitehall can communicate with asset-manager.

Depends on #163

### CODE REVIEW

File: docker/nginx.tmpl [hunk 1]
```diff
@@ -293,6 +293,12 @@ server {
     proxy_set_header Host asset-manager.dev.gov.uk;
     proxy_pass {{ trim $proto }}://{{ trim $upstream_name }};
   }
+
+  # Ref: https://github.com/alphagov/govuk-puppet/blob/2535ae6fe00ad8d91ee0fc32b7c6e9bd8e86b911/modules/govuk/templates/static_extra_nginx_config.conf.erb#L30
+  location /government/uploads/system/uploads/image_data/file/ {
+    proxy_set_header Host asset-manager.dev.gov.uk;
+    proxy_pass {{ trim $proto }}://{{ trim $upstream_name }};
+  }
```
Comment (kevindew) [inline docker/nginx.tmpl]: Thanks for digging that out 👍 

Might make sense in a separate commit

File: spec/whitehall/uploading_attachment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    expect_rendering_application("government-frontend")
+    expect_url_matches_draft_gov_uk
+    expect(page).to have_content(title)
+  end
+
+  def then_i_can_view_the_image
+    attachment_url = find(:xpath, "//img[@alt=\"#{attachment_alt_text}\"]")[:src]
+
+    # Asset manager returns a 302 to a placeholder image until the asset is available
+    reload_url_until_status_code(attachment_url, 200, keep_retrying_while: [302])
```
Comment (kevindew) [inline spec/whitehall/uploading_attachment_spec.rb]: super minor, but you don't need 302 to be in an array

File: docker/nginx.tmpl (hunk 1 above)
Comment (adrianclay) [reply docker/nginx.tmpl]: Done.
